File: backend/src/main/java/br/com/asaplog/veiculoseguro/entities/Apolice.java
package br.com.asaplog.veiculoseguro.entities;

import lombok.Data;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.math.BigDecimal;
import java.time.Instant;
import java.util.UUID;

@Data
public class Apolice {
    @Id
    private String id;
    @GeneratedValue
    private UUID codigo;
    @NotNull
    private Instant inicioVigencia;
    @NotNull
    private Instant fimVigencia;
    @NotNull
    private String placaVeiculo;
    @NotNull
    private BigDecimal valor;
    @NotNull
    @ManyToOne
    private Cliente cliente;
}

File: frontend/src/models/Apolice.ts
import { ClienteApolice } from "./ClienteApolice";

export interface Apolice {
  id?: string;
  codigo? : number,
  inicioVigencia: string,
  fimVigencia: string,
  placaVeiculo: string,
  valor: number,
  clienteCpf?: string,
}